src/components/textarea/textarea.jsx

```diff
@@ -0,0 +1,129 @@
+import PropTypes from 'prop-types';
+import React, {
+  forwardRef,
+  useEffect,
+  useRef,
+  useImperativeHandle,
+} from 'react';
+import cautionSvg from '../../svgs/caution.svg';
+import FormControl from '../form-control/form-control.jsx';
+import Icon from '../icon/icon.jsx';
+import useMounted from '../../hooks/use-mounted.js';
+import useValidation from '../../hooks/use-validation.jsx';
+import styles from './textarea.css';
+
+function getClassName(className, validationMessage) {
+  if (className) return className;
+  return validationMessage ? styles['invalid-textarea'] : styles.textarea;
+}
+
+const Textarea = forwardRef(function Textarea({
+  className,
+  cssContainer,
+  id,
+  label,
+  name,
+  onChange,
+  onBlur,
+  placeholder,
+  readOnly,
+  required,
+  validationMessage,
+  value,
+}, forwardedRef) {
+  const fallbackRef = useRef();
+  const ref = forwardedRef || fallbackRef;
+  const textareaRef = useRef();
+  const mounted = useMounted();
+  const [validationMessageValue, validate] = useValidation(validationMessage, textareaRef);
+
+  function blurHandler(event) {
+    validate();
+    onBlur(event);
+  }
+
+  function changeHandler(event) {
+    validate();
+    onChange(event);
+  }
+
+  useEffect(() => {
```

Comment:
If you're making revisions, can you check whether this should just be `useLayoutEffect` instead of `useEffect`? If so, we can remove the `mounted.current` because `useLayoutEffect` are only ran after immediately painting which only happens when the component is mounted. 

Note: we don't wanna get carried away with using `useLayoutEffect` since it will cause longer paint times. AFAIK, we only should be using it when mutating any DOM ref.